src/test/scala/mesosphere/util/state/zk/ZKStoreTest.scala

```diff
@@ -0,0 +1,124 @@
+package mesosphere.util.state.zk
+
+import java.io.File
+
+import com.twitter.zk.ZkClient
+import mesosphere.marathon.StoreCommandFailedException
+import mesosphere.marathon.integration.setup.{ IntegrationFunSuite, IntegrationTestConfig, ProcessKeeper }
+import mesosphere.util.state.PersistentEntity
+import org.apache.commons.io.FileUtils
+import org.apache.zookeeper.ZooDefs.Ids
+import org.scalatest._
+import org.scalatest.concurrent.ScalaFutures._
+
+import scala.collection.JavaConverters._
+
+class ZKStoreTest extends IntegrationFunSuite with BeforeAndAfterAll with BeforeAndAfter with Matchers {
```

Comment:
Nice Test!

It would be great if you created an `AbstractStoreTest` or something similar and then instantiated with your three different store implementations.